views/src/main/java/io/micronaut/views/thymeleaf/WebContext.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.micronaut.views.thymeleaf;
+
+import io.micronaut.http.HttpRequest;
+import org.thymeleaf.context.AbstractContext;
+
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Web-oriented implementation of the {@link org.thymeleaf.context.IContext} and {@link
+ * org.thymeleaf.context.IWebContext} interfaces.
+ *
+ * @author Semyon Gashchenko
+ */
+class WebContext extends AbstractContext {
```

Comment:
Any particular reason why everything is package-private?

Reply:
I think that class may not be useful outside this package.

Reply:
If the `EngineContextFactory` needs to be overriden, users will need access to this class. Please make it public

Reply:
Done.